aclk/aclk_query_queue.c

```diff
@@ -20,27 +20,30 @@ static struct aclk_query_queue {
 
 static inline int _aclk_queue_query(aclk_query_t query)
 {
+    now_realtime_timeval(&query->created_tv);
     query->created = now_realtime_usec();
+
     ACLK_QUEUE_LOCK;
+
+    // blocking?
     if (aclk_query_queue.block_push) {
         ACLK_QUEUE_UNLOCK;
         if(!netdata_exit)
             error("Query Queue is blocked from accepting new requests. This is normally the case when ACLK prepares to shutdown.");
         aclk_query_free(query);
         return 1;
     }
-    if (!aclk_query_queue.head) {
+
+    // insert @ tail.
+    if (likely(aclk_query_queue.head))
+        aclk_query_queue.tail->next = query;
+    else
         aclk_query_queue.head = query;
-        aclk_query_queue.tail = query;
-        ACLK_QUEUE_UNLOCK;
-        return 0;
-    }
-    // TODO deduplication
```

Comment:
why are we removing this comment if deduplication was not implemented?

Reply:
I think I misunderstood this comment to mean "deduplicate the code", which is what I ended up doing. If it means deduplicate the query based off of request content, then that is indeed not implemented.

Reply:
yes that is what the comment meant... there should be deduplication based on query content... e.g. if cloud does chart data query that we don't deliver in time and cloud does another for the same chart (with same parameters) we are supposed to throw away the old one.

Reply:
agreed the comment can be misunderstood easily... my fault (as I wrote that comment)